imports/plugins/core/accounts/server/no-meteor/mutations/removeAccountFromGroup.js

```diff
@@ -0,0 +1,77 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+import ensureRoles from "/imports/plugins/core/accounts/server/no-meteor/util/ensureRoles";
+
+const inputSchema = new SimpleSchema({
+  accountId: String,
+  groupId: String
+});
+
+/**
+ * @name accounts/removeAccountFromGroup
+ * @memberof Mutations/Accounts
+ * @summary Removes a user from a group for a shop, and adds them to the default customer group.
+ * Updates the user's permission list to reflect.
+ * (NB: At this time, a user only belongs to only one group per shop)
+ * @param {Object} context - GraphQL execution context
+ * @param {Object} input - Necessary input for mutation. See SimpleSchema.
+ * @param {String} input.accountId - decoded ID of account on which entry should be updated, for admin
+ * @param {String} input.groupId - decoded ID of the group to remove account from
+ * @returns {Promise<Object>} with modified group
+ */
+export default async function removeAccountFromGroup(context, input) {
+  inputSchema.validate(input);
+  const { appEvents, collections, userHasPermission, userId: userIdFromContext } = context;
+  const { Accounts, Groups, users } = collections;
+  const {
+    accountId: providedAccountId,
+    groupId
+  } = input;
+
+  const { shopId } = await Groups.findOne({ _id: groupId }) || {};
+
+  // we are limiting group method actions to only users with admin roles
+  // this also include shop owners, since they have the `admin` role in their Roles.GLOBAL_GROUP
+  if (!userHasPermission(["admin"], shopId)) {
+    throw new ReactionError("access-denied", "Access denied");
+  }
+
+  // make sure account exists in `Accounts` and `users` collections
+  const account = await Accounts.findOne({ _id: providedAccountId });
+  if (!account) throw new ReactionError("not-found", "No account found");
+
+  const accountUser = await users.findOne({ _id: account.userId });
+  if (!accountUser) throw new ReactionError("not-found", "No user found");
+
+  // get default roles for a customer, which is the group
+  // this user will belong to after being removed from previous group
+  const defaultCustomerGroupForShop = await Groups.findOne({ slug: "customer", shopId }) || {};
+
+  await ensureRoles(context, defaultCustomerGroupForShop.permissions);
+  await users.updateOne({
+    _id: account.userId
+  }, {
+    $set: {
+      [`roles.${shopId}`]: defaultCustomerGroupForShop.permissions
+    }
+  });
+
+  const { value: updatedAccount } = await Accounts.findOneAndUpdate({
+    _id: providedAccountId,
+    groups: groupId
+  }, {
+    $set: { "groups.$": defaultCustomerGroupForShop._id }
+  });
+
+  if (!updatedAccount) {
+    throw new ReactionError("server-error", "Unable to remove account from group");
+  }
+
+  await appEvents.emit("afterAccountUpdate", {
+    account: updatedAccount,
+    updatedBy: userIdFromContext,
+    updatedFields: ["groups"]
+  });
+
+  return defaultCustomerGroupForShop;
```

Comment:
It's a bit strange that we decided to return the group at all, and the docs are not clear about which group is returned, but to me it seems more logical to expect the removed group back rather than the new group. So I would return `group` instead of `defaultCustomerGroupForShop`.

It's actually a poorly named mutation in general, because it's actually _changing_ the account group (to "customer") rather than just removing them from a group. I guess that's because we wanted everybody to always be in exactly one group per shop. So I'd argue we don't really need this mutation. You can call `addAccountToGroup` with the "customer" group, and it will do mostly the same thing. (Even that one would be better named `setAccountGroupForShop`.)

There is also a slight difference between those two mutations, which is a bit worrying. When switching back to the "customer" group here, we simply overwrite the full `roles` list for the shop, setting it to match the customer group permissions. But in `addAccountToGroup`, we delete all roles from the previous group and then add all roles from the new group. The difference is that `addAccountToGroup` way would keep one-off roles that have been added directly to a user outside the group paradigm. I don't know if this is really what we want or if users should have only the permissions from the group they are in and no others.

All of that being said, it doesn't seem like any core code actually calls `removeAccountFromGroup`, so how about you address these comments here but then mark it deprecated in the gql schema so that we can remove it in the next major release? The roles logic in `addAccountToGroup` is fine for now but we may want to raise the question of whether extra roles should be allowed on users (fits in with the work of the identity+auth pod that is starting).